plugins/modules/apitoken.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import requests
+
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: apitoken
+
+short_description: Generate API Tokens
+
+version_added: "1.0.0"
+
+description: Generates API tokens given the OFFLINE token
+
+options:
+    offline_token:
+      description: The offline token used to generate the API Token
+      required: true
+
+author:
+    - Name (@vjayaramrh)
+"""
+
+EXAMPLES = r"""
+# Use argument
+- name: Generate API Token
+  apitoken
+    offline_token: blah
+
+"""
+
+RETURN = r"""
+apitoken:
+    description: A message with the status
+    type: string
+    returned: always
+    sample: blah
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+URL = f"https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+
+def run_module():
+    module_args = dict(
+        offline_token=dict(type="str", required=True),
+    )
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=False)
+
+    result = {}
+    # get offline_token
+    off_token = module.params.get('offline_token')
+    if off_token:
+        # Set headers
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Accept': 'application/json'
+        }
+        #set data
+        data = {}
+        data['grant_type'] = "refresh_token"
+        data['client_id'] = "cloud-services"
+        data['refresh_token'] = off_token
+
+        # Generate API token
+        response = requests.post(f"{URL}", data=data, json=None, headers=headers)
+
+        if not response.ok:
+            result = dict(changed=True, response=response.text)
+            module.fail_json(msg="Error generating token", **result)
+
+        result = dict(response.json())
```

Comment:
This will not return `apitoken` as specified in the RETURN section above. It looks like it returns a number of keys...

`curl --silent --header "Accept: application/json" --header "Content-Type: application/x-www-form-urlencoded" --data-urlencode "grant_type=refresh_token" --data-urlencode "client_id=cloud-services" --data-urlencode "refresh_token=${OFFLINE_TOKEN}" "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token" | jq 'keys'
[
  "access_token",
  "expires_in",
  "id_token",
  "not-before-policy",
  "refresh_expires_in",
  "refresh_token",
  "scope",
  "session_state",
  "token_type"
]
`
This is fine, but the documentation should be updated to reflect what the module is returning.

Reply:
@clwheel Appreciate the suggestion, will make updates.